src/App.js

```diff
@@ -36,10 +38,43 @@ class App extends Component {
     return Math.floor((new Date() - date) / (365 * 60 * 24 * 1000 * 60));
   }
 
+  handleSort = (e) => {
+    let sortBy = this.state.sortBy;
+    let newSort = e.target.innerText.toLowerCase();
+    let orderDesc = this.state.orderDesc;
+  
+    if(sortBy === newSort){
+        orderDesc = !orderDesc;
+    }else{
+      sortBy = newSort;
+      orderDesc = false;      
+    }
+    
+    this.setState({
+      orderDesc: orderDesc,
+      sortBy: sortBy
+    }, () => this.orderTechnologies());
+  }
+
+  orderTechnologies(){
+    let techList = Object.assign([], this.state.technologies);
+
+    if(this.state.sortBy === 'tech'){
+      techList.sort((a, b) => this.state.orderDesc ? a.name.toLowerCase() < b.name.toLowerCase() : a.name.toLowerCase() > b.name.toLowerCase() )
+    }else{
+      techList.sort((a, b) => this.state.orderDesc ? a.released > b.released : a.released < b.released )
+    }
```

Comment:
Actually you're supposed to return a number(1, -1, 0) from your compare function, not a boolean.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#Description